### DISCUSSION TITLE
Public page setting not working

### DISCUSSION THREAD

**Discussion by NicolasC007** [Budibase/budibase#4280] [Help] [answered]
Hi!
I have set up a MongoDB connection with a form working fine. The problem is that whenever I try to share the app to collect information from end users, it keeps asking me to log in (tried different browsers and incognito).
My insert query is already set to 'public' and so is my 'home' screen.
I've tried with different links, but still getting redirected to the login/create account url.
I'm not sure what else to do.
Any ideas?

Thanks! I really appreciate it!

**Comment by NicolasC007** [comment, marked as answer]
I was using the wrong link.
Sorry. All good now

**Reply by jeremyirawan** [reply]
can you elaborate more about your solution?

**Reply by NicolasC007** [reply]
Hi, make sure that your app is "open" and not "locked by you". You will probably need to close the dev environment. Then click on View Published App and that is the link you can use to access the public part of the app (given that you have set "public" rights to the pages you want to make public)

![image](https://user-images.githubusercontent.com/98494891/152559879-9af094b4-f0dd-4c22-903f-c95b718e28f9.png)

Cheers!
